samples/drivers/uart/echo_bot/src/main.c

```diff
@@ -78,7 +78,18 @@ void main(void)
 	}
 
 	/* configure interrupt and callback to receive data */
-	uart_irq_callback_user_data_set(uart_dev, serial_cb, NULL);
+	int ret = uart_irq_callback_user_data_set(uart_dev, serial_cb, NULL);
+
+	if (ret < 0) {
+		if (ret == -ENOTSUP) {
+			printk("Interrupt-driven UART API support not enabled\n");
+		} else if (ret == -ENOSYS) {
+			printk("UART device does not support interrupt-driven API\n");
```

Comment:
Make clear that it's the driver that doesn't support the interrupt-driven API and most probably not the hardware.

```suggestion
			printk("UART driver does not support interrupt-driven API\n");
```

Reply:
If we look at the drivers that do not support the interrupt API, it appears most of them are for virtual or uart emulating devices, like RTT or native posix.  So it really is the case that the hardware doesn't support interrupts.